app/src/main/res/layout/add_profile_activity.xml

```diff
@@ -163,22 +131,64 @@
             app:layout_constraintStart_toStartOf="parent"
             app:layout_constraintTop_toTopOf="parent"
             profile:error="@{viewModel.pinErrorMsg}" />
+
+          <androidx.constraintlayout.widget.ConstraintLayout
+            android:id="@+id/add_profile_activity_allow_download_constraint_layout"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="28dp"
+            android:layout_marginTop="28dp"
+            android:layout_marginEnd="32dp"
+            app:layout_constraintTop_toBottomOf="@+id/add_profile_activity_input_confirm_pin_profile_input_view">
+
+            <TextView
+              android:id="@+id/add_profile_activity_allow_download_heading_text_view"
+              android:layout_width="wrap_content"
+              android:layout_height="wrap_content"
+              android:fontFamily="sans-serif"
+              android:text="@string/add_profile_allow_download_heading"
+              android:textColor="@{viewModel.validPin ? @color/oppiaPrimaryTextDark : @color/grey}"
+              android:textSize="16sp"
+              app:layout_constraintStart_toStartOf="parent"
+              app:layout_constraintTop_toTopOf="parent" />
+
+            <TextView
+              android:id="@+id/add_profile_activity_allow_download_text_view"
+              android:layout_width="0dp"
+              android:layout_height="wrap_content"
+              android:fontFamily="sans-serif"
+              android:text="@string/add_profile_allow_download_sub"
+              android:textColor="@color/light_grey"
+              android:textSize="14sp"
+              app:layout_constraintStart_toStartOf="parent"
+              app:layout_constraintTop_toBottomOf="@+id/add_profile_activity_allow_download_heading_text_view"
+              app:layout_constraintWidth_percent=".8" />
+
+            <Switch
+              android:id="@+id/add_profile_activity_allow_download_switch"
```

Comment:
Give minHeight & minWidth 48dp for this switch to pass accessibility test. Ditto in landscape.

Reply:
Done.